src/snowflake/snowpark/session.py

```diff
@@ -870,81 +901,73 @@ def _resolve_packages(
 
         result_dict = (
             existing_packages_dict if existing_packages_dict is not None else {}
         )
+        unsupported_packages: List[str] = []
         for package, package_info in package_dict.items():
             package_name, use_local_version, package_req = package_info
             package_version_req = package_req.specs[0][1] if package_req.specs else None
 
             if validate_package:
-                unavailable_pkg_err_msg = (
-                    "it is not available in Snowflake. Check information_schema.packages "
-                    "to see available packages for UDFs. If this package is a "
-                    '"pure-Python" package, you can find the directory of this package '
-                    "and add it via session.add_import(). See details at "
-                    "https://docs.snowflake.com/en/developer-guide/snowpark/python/creating-udfs.html#using-third-party-packages-from-anaconda-in-a-udf."
-                )
-
-                if package_name not in valid_packages:
-                    is_anaconda_terms_acknowledged = self._run_query(
-                        "select system$are_anaconda_terms_acknowledged()"
-                    )[0][0]
-                    if is_anaconda_terms_acknowledged:
-                        detailed_err_msg = unavailable_pkg_err_msg
-                    else:
-                        detailed_err_msg = (
-                            "Anaconda terms must be accepted by ORGADMIN to use "
-                            "Anaconda 3rd party packages. Please follow the instructions at "
-                            "https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-packages.html#using-third-party-packages-from-anaconda."
-                        )
-
-                    raise ValueError(
-                        f"Cannot add package {package_name} because {detailed_err_msg}"
-                    )
-                elif package_version_req and not any(
-                    v in package_req for v in valid_packages[package_name]
+                if package_name not in valid_packages or (
+                    package_version_req
+                    and not any(v in package_req for v in valid_packages[package_name])
                 ):
-                    raise ValueError(
-                        f"Cannot add package {package_name}=={package_version_req} because {unavailable_pkg_err_msg}"
+                    version_text = (
+                        f"(version {package_version_req})"
+                        if package_version_req is not None
+                        else ""
                     )
+                    if is_in_stored_procedure():  # pragma: no cover
+                        raise RuntimeError(
+                            f"Cannot add package {package_name}{version_text} because it is not present on Anaconda and "
+                            f"cannot be installed via Pip as you are executing this code inside a stored procedure."
```

Comment:
```suggestion
                            f"Cannot add package {package_name}{version_text} because it is not available in Snowflake and "
                            f"cannot be installed via Pip as you are executing this code inside a stored procedure. "
```

Can you re-add the part which tells users to check `information_schema.packages` and suggests different ways to upload pure-python packages.